### PR TITLE
fix: improve dock center space calculation and taskmanager layout

### PR DESCRIPTION
**Pull request by wjyrich**
1. Replaced temporary overflow logic with dynamic calculation of remaining space for center dock area
2. Added function to count center applets excluding TaskManager for accurate space allocation
3. Simplified dockCenterPart implicit dimensions by removing TaskManager-specific adjustments
4. Updated TaskManager to use consistent remaining space calculation and improved implicit dimension logic
5. Enhanced app container sizing to properly distribute available space among app items
6. Fixed dockItemMaxSize calculation to use correct remaining space reference

Log: Improved dock layout calculation for better space distribution and task manager display

Influence:
1. Test dock layout in different positions (left, right, top, bottom)
2. Verify center area plugins display correctly with varying numbers
3. Check TaskManager app item sizing and spacing in different configurations
4. Test with multiple center area plugins alongside TaskManager
5. Verify layout stability when adding/removing center plugins
6. Test edge cases with minimal/maximal numbers of app items

fix: 改进任务栏中心区域空间计算和任务管理器布局

1. 用动态计算替换临时的溢出逻辑，准确计算中心区域剩余空间
2. 新增函数统计除任务管理器外的中心区域插件数量，实现精确空间分配
3. 简化dockCenterPart的隐式尺寸计算，移除任务管理器特定调整
4. 更新任务管理器使用一致的剩余空间计算和改进的隐式尺寸逻辑
5. 增强应用容器尺寸计算，在应用项之间合理分配可用空间
6. 修复dockItemMaxSize计算，使用正确的剩余空间引用

Log: 改进任务栏布局计算，优化空间分配和任务管理器显示

Influence:
1. 测试任务栏在不同位置（左、右、上、下）的布局
2. 验证中心区域插件在不同数量下的正确显示
3. 检查任务管理器应用项在不同配置下的尺寸和间距
4. 测试多个中心区域插件与任务管理器共存的情况
5. 验证添加/移除中心插件时的布局稳定性
6. 测试最小/最大应用项数量的边界情况

PMS: BUG-355833

## Summary by Sourcery

Improve dock center area space calculation and TaskManager layout for more accurate and consistent space distribution.

Enhancements:
- Compute remaining dock center space dynamically, excluding the TaskManager and other center applets’ footprints for more accurate layout.
- Simplify dock center container sizing by relying directly on the center loader’s implicit dimensions.
- Align TaskManager sizing with the shared remaining-space calculation and adjust its implicit width/height based on alignment and orientation.
- Constrain TaskManager app item sizes using remaining space and item count to distribute available space evenly.
- Base dockItemMaxSize on the unified remaining-center-space property for consistent item sizing.

### CODE REVIEW

File: panels/dock/package/main.qml [hunk 1]
```diff
@@ -19,13 +19,33 @@ Window {
...
+            }
+        }
+        otherCenterAppletsCount = count;
+    }
+
+    property int dockRemainingSpaceForCenter: {
+        let otherCount = otherCenterAppletsCount;
+        
+        let otherHeight = otherCount > 0 ? (otherCount * (useColumnLayout ? dockItemMaxSize : dock.dockSize) + otherCount * gridLayout.rowSpacing) : 0;
+        let otherWidth = otherCount > 0 ? (otherCount * (useColumnLayout ? dock.dockSize : dockItemMaxSize) + otherCount * gridLayout.columnSpacing) : 0;
```
Comment (BLumia) [inline panels/dock/package/main.qml]: 这两个总有一个会是不会被用到的，是不是不用三目运算符读起来反而会清晰一些，还少（虽然只有）一行多余的计算。

**Review by 18202781743 — DISMISSED** [review]

File: panels/dock/package/main.qml [hunk 2]
```diff
@@ -19,13 +19,38 @@ Window {
...
+        otherCenterAppletsCount = count;
+    }
+
+    property int dockRemainingSpaceForCenter: {
+        const otherCount = otherCenterAppletsCount;
+        const spacing = useColumnLayout ? gridLayout.rowSpacing : gridLayout.columnSpacing;
+            
+        let otherOccupied = 0;
+        if (otherCount > 0) {
+           otherOccupied = otherCount * dockItemMaxSize + otherCount * spacing;
```
Comment (BLumia) [inline panels/dock/package/main.qml]:
```suggestion
            otherOccupied = otherCount * dockItemMaxSize + otherCount * spacing;
```

**Review by 18202781743 — DISMISSED** [review]

File: panels/dock/package/main.qml [hunk 3]
```diff
@@ -460,6 +470,9 @@ Window {
         // TODO missing property binding to update ProxyModel's filter and sort opearation.
         Repeater {
             model: Applet.appletItems
+            onCountChanged: {
```
Comment (18202781743) [inline panels/dock/package/main.qml]: 这个也不需要了吧，

**Review by 18202781743 — APPROVED** [review]
